stravalib/model.py

```diff
@@ -432,379 +553,165 @@ class ActivityKudos(Athlete):
-    average_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Average speed for lap
-    max_speed = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-        units=uh.meters_per_second,
-    )  #: Max speed for lap
-    average_cadence = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average cadence for lap
-    average_watts = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average watts for lap
-    average_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Average heartrate for lap
-    max_heartrate = Attribute(
-        float,
-        (
-            SUMMARY,
-            DETAILED,
-        ),
-    )  #: Max heartrate for lap
-    lap_index = Attribute(int, (SUMMARY, DETAILED))  #: Index of lap
-    device_watts = Attribute(
-        bool, (SUMMARY, DETAILED)
-    )  # true if the watts are from a power meter, false if estimated
-    pace_zone = Attribute(int, (SUMMARY, DETAILED))  #: (undocumented)
-    split = Attribute(int, (SUMMARY, DETAILED))  #: Split number
+    _naive_local = validator('start_date_local', allow_reuse=True)(naive_datetime)
 
+    _activity_extension = extend_types('activity', model_class='Activity')
+    _athlete_extension = extend_types('athlete', model_class=Athlete)
 
-class Map(IdentifiableEntity):
-    id = Attribute(str, (SUMMARY, DETAILED))  #: Alpha-numeric identifier
-    polyline = Attribute(str, (SUMMARY, DETAILED))  #: Google polyline encoding
-    summary_polyline = Attribute(
-        str, (SUMMARY, DETAILED)
-    )  #: Google polyline encoding for summary shape
+
+class Map(PolylineMap):
+    pass
 
 
-class Split(BaseEntity):
+class Split(Split, BackwardCompatibilityMixin, DeprecatedSerializableMixin):
     """
     A split -- may be metric or standard units (which has no bearing
     on the units used in this object, just the binning of values).
     """
 
-    distance = Attribute(float, units=uh.meters)  #: Distance for this split
-    elapsed_time = (
-        TimeIntervalAttribute()
-    )  #: :class:`datetime.timedelta` of elapsed time for split
-    elevation_difference = Attribute(
-        float, units=uh.meters
-    )  #: Elevation difference for split
-    moving_time = (
-        TimeIntervalAttribute()
-    )  #: :class:`datetime.timedelta` of moving time for split
-    average_heartrate = Attribute(float)  #: Average HR for split
-    split = Attribute(int)  #: Which split number
-    pace_zone = Attribute(int)  #: (undocumented)
-    average_speed = Attribute(float, units=uh.meters_per_second)
-    average_grade_adjusted_speed = Attribute(float, units=uh.meters_per_second)
+    _unit_registry = {
+        'distance': 'meters',
```

Comment:
similar to above - i see several items removed from this object such as moving time. were those items dropped from the API?

Reply:
same question for below - things seemed to be removed. many thanks

Reply:
I hope the above answers about inheritance from the generated model explain enough. If not, please let me know.

Reply:
it does. i see how i was really asking the same question over and over! appreciate your patience!!